### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by BenAB98**

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -1,9 +1,17 @@
 <script setup>
-
+import MainPage from './pages/MainPage.vue'
+// import Edit from './pages/Edit.vue';
```
Comment (amRashidi) [inline src/App.vue]: remove comments

File: src/components/ButtonPrimary.vue [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+})
+</script>
+
+<template>
+  <button class="rounded-lg bg-orange-500 text-white w-1/2 md:w-36 hover:scale-95 transition-all">
+    <span class="px-4 py-2 inline-block">{{ title }}</span>
+  </button>
+</template>
+
+<style scoped></style>
```
Comment (amRashidi) [inline src/components/ButtonPrimary.vue]: remove empty style tag

File: src/components/FooterComponent.vue [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      <div class="mb-4 lg:pr-2">
+        <h3 class="text-grayWhite mb-4 text-lg">About-Us</h3>
+        <p class="text-grayLight">
+          Lorem ipsum dolor sit amet consectetur adipisicing elit. Ducimus a voluptatum eius.
+          Fugiat, tempora ad!
+        </p>
+      </div>
+      <div class="text-grayLight mb-4">
+        <h3 class="text-grayWhite mb-4 text-lg">Recent Review</h3>
+        <a href="#" rel="nofallow" class="block w-fit hover:text-ogangeLink transition-all"
```
Comment (amRashidi) [inline src/components/FooterComponent.vue]: use v-for

File: src/components/FooterComponent.vue [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+          >Facebook</a
+        >
+        <a href="#" rel="nofallow" class="block w-fit hover:text-ogangeLink transition-all"
+          >Twitter</a
+        >
+        <a href="#" rel="nofallow" class="block w-fit hover:text-ogangeLink transition-all"
+          >Google+</a
+        >
+        <a href="#" rel="nofallow" class="block w-fit hover:text-ogangeLink transition-all"
+          >Pinterest</a
```
Comment (amRashidi) [inline src/components/FooterComponent.vue]: same

File: src/components/HeaderComponent.vue [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+            stroke-linecap="round"
+            stroke-linejoin="round"
+            d="M3.75 6.75h16.5M3.75 12h16.5m-16.5 5.25h16.5"
+          />
+        </svg>
+      </button>
+      <ul class="hidden lg:flex text-black justify-center items-center">
+        <li class=""><a href="#" class="text-ogangeLink transition-all px-4">Home</a></li>
+        <li class="border-l">
+          <a
```
Comment (amRashidi) [inline src/components/HeaderComponent.vue]: vfor
href???

File: src/components/MainComponent.vue [hunk 6]
```diff
@@ -0,0 +1,30 @@
+<script setup>
+import { ref } from 'vue'
+import { fetchApi } from '../fetchAPI'
+import Movies from './Movies.vue'
+import Skeleton from './Skeleton.vue'
+const moviesList = ref([])
+fetchApi('https://api.themoviedb.org/3/movie/top_rated?language=en-US&page=3')
```
Comment (amRashidi) [inline src/components/MainComponent.vue]: base url

File: src/components/MainComponent.vue [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  .then((response) => (moviesList.value = response.results))
+  .catch((err) => console.log(err))
+</script>
+
+<template>
+  <main
+    class="border-b-2 rounded-t-md bg-white p-7 pb-10 flex flex-col gap-y-7 container md:max-w-screen-md lg:max-w-[1170px] md:grid md:grid-cols-2 md:gap-x-7 lg:grid-cols-4 lg:grid-rows-3 lg:max-h-[825px]"
+  >
+    <template v-for="number in 4" :key="number">
+      <skeleton v-if="!moviesList.length" height="100vh" width="100%" borderRadius="12px" />
```
Comment (amRashidi) [inline src/components/MainComponent.vue:26]: wrong condition

File: src/components/MainComponent.vue [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    </template>
+    <template v-for="(movie, index) in moviesList" :key="movie.id">
+      <movies
+        v-if="index < 7"
+        :index="index"
+        :alt="movie.title"
+        :src="`https://image.tmdb.org/t/p/w500${movie.poster_path}`"
```
Comment (amRashidi) [inline src/components/MainComponent.vue]: base url for images

File: src/fetchAPI.js [hunk 9]
```diff
@@ -0,0 +1,16 @@
+export async function fetchApi(url) {
+  const options = {
+    method: 'GET',
+    headers: {
+      accept: 'application/json',
+      'content-type': 'application/json',
+      Authorization: `Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiIwM2E1YTEzZTk4NTk4NDcyNTRjYWQ5M2E5Nzg2MjEzYyIsInN1YiI6IjY0NzFiMmE0YmUyZDQ5MDBiZjlkNWFhMSIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.XuVUHSkSjI6SYi-p-Mf-32ZvVklBRRin8sFCmiNqBaE`
```
Comment (amRashidi) [inline src/fetchAPI.js]: avoid hardcoding your token!!
do this with env

File: src/layout/ScrollingFree.vue [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+import ScrollingMovieCards from '../components/ScrollingMovieCards.vue'
+import { fetchApi } from '../fetchAPI'
+defineProps({
+  title: {
+    type: String,
+    required: true
+  }
+})
+const moviesList = ref([])
+fetchApi('https://api.themoviedb.org/3/movie/popular?language=en-US&page=2')
```
Comment (amRashidi) [inline src/layout/ScrollingFree.vue]: base url

File: src/layout/ScrollingPapulars.vue [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+import ScrollingMovieCards from '../components/ScrollingMovieCards.vue'
+import { fetchApi } from '../fetchAPI'
+defineProps({
+  title: {
+    type: String,
+    required: true
+  }
+})
+const moviesList = ref([])
+fetchApi('https://api.themoviedb.org/3/movie/popular?language=en-US&page=1')
```
Comment (amRashidi) [inline src/layout/ScrollingPapulars.vue]: base url

File: src/pages/Edit.vue [hunk 12]
```diff
@@ -0,0 +1,144 @@
+<script setup>
+import FormComponent from '../components/FormComponent.vue'
+</script>
+
+<template>
+  <div class="bg-background">
+    <header-component />
+    <form-component title="Edit Profile">
+      <div class="border flex items-center p-2 rounded-lg">
+        <span class="">
```
Comment (amRashidi) [inline src/pages/Edit.vue]: v-for

File: src/pages/Login.vue [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+          class="p-2 w-full"
+          autocomplete="off"
+          placeholder="Password..."
+        />
+      </div>
+      <div class="flex flex-col items-center gap-y-3">
+        <a href="#" class="text-grayLight hover:text-ogangeLink transition-all w-fit text-center"
+          >Forgot Your Password?</a
+        >
+        <a href="signup.html" class="text-grayLight hover:text-ogangeLink transition-all w-fit"
```
Comment (amRashidi) [inline src/pages/Login.vue]: href!!

File: src/pages/MainPage.vue [hunk 14]
```diff
@@ -0,0 +1,17 @@
+<script setup>
+import MainComponent from '../components/MainComponent.vue'
```
Comment (amRashidi) [inline src/pages/MainPage.vue]: global address

File: src/pages/MovieDetailsPage.vue [hunk 15]
```diff
@@ -0,0 +1,16 @@
+<script setup>
+import { fetchApi } from '../fetchAPI'
+
+fetchApi('https://api.themoviedb.org/3/configuration')
```
Comment (amRashidi) [inline src/pages/MovieDetailsPage.vue]: base url

File: src/pages/SignUp.vue [hunk 16]
```diff
@@ -0,0 +1,150 @@
...
+          placeholder="Full Name..."
+        />
+      </div>
+    </form-component>
+    <footer-component />
+  </div>
+</template>
+
+<script setup>
+import FormComponent from '../components/FormComponent.vue'
```
Comment (amRashidi) [inline src/pages/SignUp.vue]: global address url

**Review by amRashidi — CHANGES_REQUESTED** [review]

**Comment by BenAB98** [comment]
@amRashidi Changes have been made as requested

**Review by amRashidi — APPROVED** [review]

**Pull request merged**